samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaConsumerProxy.java

```diff
@@ -51,7 +53,7 @@
  * This class is not thread safe. There will be only one instance of this class per KafkaSystemConsumer object.
  * We still need some synchronization around kafkaConsumer. See pollConsumer() method for details.
  */
-class KafkaConsumerProxy<K, V> {
+public class KafkaConsumerProxy<K, V> {
```

Comment:
Do the K, V type parameters matter for this class (and the underlying KafkaConsumer)?. If not, maybe remove for simplicity.

Reply:
The <K, V> does get used for the kafka `Consumer` which is wired in. Ultimately, it doesn't get used since `IncomingMessageEnvelope` doesn't have type params, but it looks like it helps to ensure the wiring matches up for the proxy and Samza/Kafka consumer objects.

Reply:
I'd prefer to remove the types from the Consumer as well, since they're mostly meaningless. But we can do it in a separate PR.